### DISCUSSION TITLE
Completing the implementation of the REST API

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#1855] [Ideas]
When it comes to integrating an ERP system with Magento 1, one is forced to use the SOAP API which is old and slow, to say the least...Magento 1 has had a few REST API endpoints for a long time but it's missing a lot of essential functionality. I think a lot of people would benefit from a fully implemented REST API that covers all the features and functionality available in the SOAP API. What are your opinions?

**Comment by kiatng** [comment]
I needed to interface with another server in a current project. The other server uses REST, as a client, I use [Guzzle](https://docs.guzzlephp.org/en/stable/) in OM, [here's how I installed it](https://makandracards.com/magento1/500719-how-to-add-guzzle-package-using-composer). But I managed to persuade the stakeholders that we use [JSON-RPC](https://www.jsonrpc.org/specification), which can be readily extended in OM. It inherits existing features of users and roles management from SOAP and XML-RPC. I also managed to figure  out a way to call the JSON-RPC API similar in style to SOAPv2, in which the client call the method directly instead of the `call` method. If JSON-RPC is of interest to the OM community, I can make a PR.
